vanguard_driver.py

```diff
@@ -377,21 +382,27 @@ def run(args: VanguardDriverOpts, lang: Platform):
 
     args.logger.info(f"----Running Vanguard with {args.detector} detector----")
 
+    to_dir: Callable[[Path], Path] = lambda path: path if path.is_dir() else path.parent
+    get_app_dir: Callable[[Path], str] = lambda path: str(to_dir(path).absolute())
+    application_dirs: List[str] = list(map(get_app_dir, args.src_paths))
```

Comment:
nit: `get_app_dir` seems unnecessarily verbose to me, I would simplify this to
```suggestion
    to_dir: Callable[[Path], Path] = lambda path: path if path.is_dir() else path.parent
    application_dirs: List[str] = [str(to_dir(path).absolute()) for path in args.src_paths]
```
Also, since the `application_dirs` are only used to build the Vanguard command, it's fine to leave them as `Path`s.

Reply:
Simplified in https://github.com/Veridise/Vanguard/pull/83/commits/ee87a6f9fc2bb646ba0ec6318f5bf5b37b277e48